app/models/pame_evaluation.rb

```diff
@@ -1,7 +1,350 @@
+      {
+        current_page: evaluations.current_page,
+        per_page: evaluations.per_page,
+        total_entries: evaluations.total_entries,
+        total_pages: evaluations.total_pages,
+        id: evaluation.id,
+        wdpa_id: wdpa_id,
+        restricted: evaluation.restricted,
+        iso3: iso3,
+        methodology: evaluation.methodology,
+        year: evaluation.year.to_s,
+        url: evaluation.url,
+        metadata_id: evaluation.metadata_id,
+        source_id: evaluation.pame_source&.id,
+        name: name,
+        designation: designation,
+        data_title: evaluation.pame_source&.data_title,
+        resp_party: evaluation.pame_source&.resp_party,
+        language: evaluation.pame_source&.language,
+        source_year: evaluation.pame_source&.year
+      }
+    end
+  end
+
+  def self.sources_to_json
+    sources = PameSource.all.order(id: :asc)
+    sources.to_a.map! do |source|
+      {
+        id: source.id,
+        data_title: source.data_title,
+        resp_party: source.resp_party,
+        year: source.year,
+        language: source.language
+      }
+    end.to_json
+  end
+
+  def self.filters_to_json
+    unique_methodologies = PameEvaluation.pluck(:methodology).uniq.sort
+    unique_iso3 = Country.pluck(:iso_3).uniq.sort
+    unique_year = PameEvaluation.pluck(:year).uniq.map(&:to_s).sort
+
+    [
+      {
+        name: 'methodology',
+        title: 'Methodology',
+        options: unique_methodologies,
+        type: 'multiple'
+      },
+      {
+        name: "iso3",
+        title: "Country",
+        options: unique_iso3,
+        type: 'multiple'
+      },
+      {
+        name: "year",
+        title: "Year of assessment",
+        options: unique_year,
+        type: 'multiple'
+      }
+    ].to_json
+  end
+
+SELECT_STATEMENT = [
+    'DISTINCT pame_evaluations.id AS assessment_id', 'pame_evaluations.metadata_id AS metadata_id',
+    'pame_evaluations.url AS url', 'pame_evaluations.year AS year',
+    'pame_evaluations.methodology AS methodology', 'pame_evaluations.wdpa_id AS wdpa_id',
+    'pame_sources.data_title AS source_data_title', 'pame_sources.resp_party AS source_resp_party',
+    'pame_sources.year AS source_year', 'pame_sources.language AS source_language',
+    'pame_evaluations.protected_area_id AS protected_area_id', 'protected_areas.name AS protected_area_name',
+    'designations.name AS designation', 'ARRAY_TO_STRING(ARRAY_AGG(countries.iso_3),\';\') AS countries'
+].freeze
+
+GROUP_BY = "pame_evaluations.id, protected_areas.wdpa_id, protected_areas.name, designation, pame_sources.data_title,
+            pame_sources.resp_party, pame_sources.year, pame_sources.language".freeze
+
+  def self.generate_csv(where_statement, restricted_where_statement)
+    where_statement = where_statement.empty? ? '' : "WHERE #{where_statement}"
+    restricted_where_statement = restricted_where_statement.empty? ? '' : "WHERE #{restricted_where_statement}"
+    query = <<-SQL
```

Comment:
Could we make this a separate private method at least? So to clean the code a little bit